tests/component/test_task_events.py

```diff
@@ -162,51 +169,96 @@ def test___done_and_every_n_samples_events_registered___run_finite_acquisition__
     assert len(every_n_samples_event_observer.events) == 10
 
 
-@pytest.mark.library_only
 def test___done_and_every_n_samples_events_registered___run_multiple_finite_acquisitions___callbacks_invoked(
     ai_task: nidaqmx.Task,
 ) -> None:
+    num_acquisitions = 3
     done_event_observer = DoneEventObserver()
+    every_n_event_count = 10
     every_n_samples_event_observer = EveryNSamplesEventObserver()
     ai_task.register_done_event(done_event_observer.handle_done_event)
     ai_task.register_every_n_samples_acquired_into_buffer_event(
         100, every_n_samples_event_observer.handle_every_n_samples_event
     )
-    ai_task.timing.cfg_samp_clk_timing(rate=10000.0, samps_per_chan=1000)
+    ai_task.timing.cfg_samp_clk_timing(
+        rate=10000.0, sample_mode=AcquisitionType.FINITE, samps_per_chan=1000
+    )
 
-    for _ in range(3):
+    for _ in range(num_acquisitions):
         ai_task.start()
         done_event_observer.wait_for_events()
-        every_n_samples_event_observer.wait_for_events(10)
+        every_n_samples_event_observer.wait_for_events(every_n_event_count)
         ai_task.stop()
 
-    assert len(done_event_observer.events) == 3
-    assert len(every_n_samples_event_observer.events) == 30
+    assert len(done_event_observer.events) == num_acquisitions
+    assert len(every_n_samples_event_observer.events) == num_acquisitions * every_n_event_count
 
 
-@pytest.mark.library_only
+@pytest.mark.xfail(reason="AB#2395984: GrpcStubInterpreter does not unregister events with DAQmx")
+@pytest.mark.library_only  # Test hangs with gRPC
```

Comment:
It doesn't run on gRPC, so why is it expected to fail?

Reply:
It's expected to fail with gRPC, but it fails by hanging, so I also added library_only. 

I don't think these marks are expressive or accurate enough. We should add `grpc_skip(reason)` and `grpc_xfail(reason)`.

Reply:
I'll address this in a separate PR.